lib/puppet/type/cisco_interface.rb

```diff
@@ -698,6 +702,206 @@ def is_to_s(value)
     end
   end # property stp_vlan_port_priority
 
+  ###########################
+  # private vlan attributes #
+  ###########################
+
+  newproperty(:switchport_mode_private_vlan_host) do
+    desc 'Switchport private host mode of the interface.'
+
+    newvalues(
+      :host,
+      :promiscuous,
+      :disabled)
+  end # property switchport_mode_private_vlan_host
+
+  newproperty(:switchport_mode_private_vlan_host_association, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc 'An array of [primary_vlan, secondary_vlan] pair. '\
+         "Valid values match format #{format} with primary_vlan "\
+         'and secondary_vlan as integer string.'
+    match_error = "must be of format #{format} with primary_vlan and "\
+                  'secondary_vlan as integer string.'
+
+    validate do |value|
```

Comment:
Not sure how to address this since split is a little generic and the input can be '10' or '10-12' or
'10,12-14,16,18-22' etc...This is comes down to an email I sent last week asking to which extent we should restrict/allow user input and what kind of check should be done on the input..